runtime/runtime/src/actions.rs

```diff
@@ -883,7 +873,6 @@ pub(crate) fn check_actor_permissions(
             }
         }
         Action::CreateAccount(_) | Action::FunctionCall(_) | Action::Transfer(_) => (),
-        #[cfg(feature = "protocol_feature_nep366_delegate_action")]
         Action::Delegate(_) => (),
```

Comment:
Just wondering, are there any corner cases to cover here? E.g. is delegating with `actor_id == account_id` legal?

Reply:
Yes, legal. The NEP didn't specify this as an error condition and the authors implemented it this way in the reference implementation.

Does it make sense? Different question. I don't see why you would want to delegate to yourself. Maybe for testing, could be interesting to test a function access key without using its allowance? But that's not really why it is implemented this way, pretty sure it was just the path of least resistance.

I don't see any other corner cases.

Reply:
Sounds good, I was just checking that we haven't missed any checks here or when writing the NEP.

I also think delegation to self only makes sense for testing, but other than a weird corner-case I don't think it leads to any problems.